rspace/src/test/scala/coop/rchain/rspace/test/InMemoryStore.scala

```diff
@@ -2,185 +2,186 @@ package coop.rchain.rspace.test
 
-  private[rspace] def putCs(txn: T, channels: Seq[C]): Unit =
-    _keys.update(hashChannels(channels), channels)
+  private[rspace] type T = Unit
 
-  private[rspace] def getChannels(txn: T, s: H) =
-    _keys.getOrElse(s, Seq.empty[C])
+  private[rspace] override def hashChannels(channels: Seq[C]): H =
+    Codec[Seq[C]]
+      .encode(channels)
+      .map((bitVec: BitVector) => Blake2b256Hash.create(bitVec.toByteArray))
+      .get
 
-  private[rspace] def createTxnRead(): Unit = ()
+  private[rspace] override def createTxnRead(): Unit = ()
 
-  private[rspace] def createTxnWrite(): Unit = ()
+  private[rspace] override def createTxnWrite(): Unit = ()
 
-  private[rspace] def withTxn[R](txn: T)(f: T => R): R =
+  private[rspace] override def withTxn[R](txn: T)(f: T => R): R =
     f(txn)
 
-  private[rspace] def collectGarbage(txn: T,
-                                     channelsHash: H,
-                                     dataCollected: Boolean = false,
-                                     waitingContinuationsCollected: Boolean = false,
-                                     joinsCollected: Boolean = false): Unit =
-    collectGarbage(channelsHash)
-
-  private[this] def collectGarbage(key: H): Unit = {
-    val as = _data.get(key).exists(_.nonEmpty)
-    if (!as) {
-      //we still may have empty list, remove it as well
-      _data.remove(key)
-    }
+  private[rspace] override def getChannels(txn: T, s: H) =
+    _dbGNATs.get(s).map(_.channels).getOrElse(Seq.empty)
 
-    val psks = _waitingContinuations.get(key).exists(_.nonEmpty)
-    if (!psks) {
-      //we still may have empty list, remove it as well
-      _waitingContinuations.remove(key)
-    }
+  private[rspace] override def getData(txn: T, channels: Seq[C]): Seq[Datum[A]] =
+    _dbGNATs.get(hashChannels(channels)).map(_.data).getOrElse(Seq.empty[Datum[A]])
 
-    val cs    = _keys.getOrElse(key, Seq.empty[C])
-    val joins = cs.size == 1 && _joinMap.contains(cs.head)
+  private[this] def getWK(txn: T, curr: Seq[C]): Seq[WaitingContinuation[P, K]] =
+    _dbGNATs
+      .get(hashChannels(curr))
+      .map(_.wks)
+      .getOrElse(Seq.empty[WaitingContinuation[P, K]])
 
-    if (!as && !psks && !joins) {
-      _keys.remove(key)
-    }
-  }
+  private[rspace] override def getWaitingContinuation(
+      txn: T,
+      curr: Seq[C]): Seq[WaitingContinuation[P, K]] =
+    getWK(txn, curr)
+      .map { (wk: WaitingContinuation[P, K]) =>
+        wk.copy(continuation = InMemoryStore.roundTrip(wk.continuation))
+      }
 
-  private[rspace] def putDatum(txn: T, channels: Seq[C], datum: Datum[A]): Unit = {
-    val key = hashChannels(channels)
-    putCs(txn, channels)
-    val datums = _data.getOrElseUpdate(key, Seq.empty[Datum[A]])
-    _data.update(key, datum +: datums)
+  override def getPatterns(txn: T, channels: Seq[C]): Seq[Seq[P]] =
+    getWK(txn, channels).map(_.patterns)
+
+  private[rspace] override def getJoin(txn: T, c: C): Seq[Seq[C]] =
```

Comment:
I think it would be good to be consistent and use the full parameter name `channel` instead of `c`.  Same with `channels` instead of `c`, 'channelsHash` for `key`, etc...

I'm not sure that I have a real preference for verbosity in naming, but consistency...yes.